src/main/java/org/apache/sysds/runtime/frame/data/columns/RaggedArray.java

```diff
@@ -49,207 +49,261 @@ public class RaggedArray<T> extends Array<T> {
-		throw new NotImplementedException("Unimplemented method 'set'");
+		if (index < _a._size)
+			_a.set(index, value);
 	}
 
 	@Override
 	public void setFromOtherType(int rl, int ru, Array<?> value) {
-		throw new NotImplementedException("Unimplemented method 'setFromOtherType'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			_a.setFromOtherType(rl, ru, value);
 	}
 
 	@Override
 	public void set(int rl, int ru, Array<T> value) {
-		throw new NotImplementedException("Unimplemented method 'set'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			if(value instanceof RaggedArray)
+				_a.set(rl, ru, ((RaggedArray) value).getInnerArray());
+			else if(_a.getClass() == value.getClass())
+				_a.set(rl, ru, value);
+			else
+				throw new RuntimeException("RaggedArray set: value type should be same to RaggedArray type " + _a.getClass());
 	}
 
+
 	@Override
 	public void set(int rl, int ru, Array<T> value, int rlSrc) {
-		throw new NotImplementedException("Unimplemented method 'set'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			if(value instanceof RaggedArray)
+				_a.set(rl, ru, ((RaggedArray) value).getInnerArray(), rlSrc);
+			else if(_a.getClass() == value.getClass())
+				_a.set(rl, ru, value, rlSrc);
+			else
+				throw new RuntimeException("RaggedArray set: value type should be same to RaggedArray type " + _a.getClass());
+
 	}
 
 	@Override
 	public void setNz(int rl, int ru, Array<T> value) {
-		throw new NotImplementedException("Unimplemented method 'setNz'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			_a.setNz(rl, ru, value);
 	}
 
 	@Override
 	public void setFromOtherTypeNz(int rl, int ru, Array<?> value) {
-		throw new NotImplementedException("Unimplemented method 'setFromOtherTypeNz'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			_a.setFromOtherTypeNz(rl, ru, value);
 	}
 
 	@Override
 	public void append(String value) {
-		throw new NotImplementedException("Unimplemented method 'append'");
+		_a.append(value);
+		super._size += 1;
 	}
 
 	@Override
 	public void append(T value) {
-		throw new NotImplementedException("Unimplemented method 'append'");
+		_a.append(value);
+		super._size += 1;
 	}
 
 	@Override
 	public Array<T> append(Array<T> other) {
-		throw new NotImplementedException("Unimplemented method 'append'");
+		super._size += other._size;
+		return _a.append(other);
 	}
 
 	@Override
 	public Array<T> slice(int rl, int ru) {
-		throw new NotImplementedException("Unimplemented method 'slice'");
+		if(rl >= 0 && rl < _a._size && ru < _a._size)
+			return _a.slice(rl, ru);
+		else if(rl >= 0 && ru >= _a._size )
+			return _a.slice(rl, _a._size - 1);
+		return null;
 	}
 
 	@Override
 	public void reset(int size) {
-		throw new NotImplementedException("Unimplemented method 'reset'");
+		_a.reset(size);
+		super._size = size;
 	}
 
 	@Override
 	public byte[] getAsByteArray() {
-		throw new NotImplementedException("Unimplemented method 'getAsByteArray'");
+		return _a.getAsByteArray();
```

Comment:
throw some exception here. this will not work, since the caller expects the byte array to be certain m size * nBytes per value.

Reply:
Then, I believe that optional should also throw an exception to be consistent?
Since currently it is not the case:
```
public byte[] getAsByteArray() {
     // technically not correct.
    return _a.getAsByteArray();
}
```